### ISSUE TITLE
New tab shows localhost:6571 when tab is switched quickly

### ISSUE THREAD

**Issue by srirambv** [brave/brave-ios#438]
### Description: 
New tab shows localhost:6571 when tab is switched quickly

### Steps to Reproduce 
  1. Open any site and quickly tap on the new tab button
  2. Tab shows localhost:6571
  3. Switch back shows page title

**Actual result:** 
New tab shows localhost:6571 when tab is switched quickly. Doesn't look like the fix #276 by @kylehickinson missed this scenario

**Expected result:**
Should just retain new tab until page load paints on the new tab

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easy

**Brave Version:** 
1.7(18.11.21.22)

**Device details:** 
iPhone 7+ (iOS 12.1)

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)? 

### Additional Information 
#249 #147
cc: @tmancey @kylehickinson

**Comment by kjozwiak** [comment]
Still reproducible using `1.22 (20.12.2.1)`. Easiest way to reproduce is opening a favourite via the context menu. STR:

* long tap on a favourite that's appearing under `brave://newtab`
* select `Open in New Tab`

You'll notice that the tab that's created will be using `localhost` for a few seconds before loading `title` of the website.

**Comment by iccub** [comment]
Should not happen anymore, will reopen if it still does

**Comment by dcm** [comment]
I just saw this happen to tabs that were set to reader view ~6 hours ago.

The screenshot below is of the iOS tabs overview on iPhone 12 Pro Max (Brave iOS 1.36): 
1. This tab was opened to reader view ~6 hours ago and has not been reactivated
2. This tab was opened and set to reader view ~6 hours ago and had the same (visible) title as (1) before it was reactivated ~10 min ago
3. Same thing as (2)

When I reactivate tab (1), after I took the screenshot, the title changes to the expected page title. The address in the URL bar at all times shows the correct URL (i.e. it doesn't say localhost:6571). Th 

So this isn't identical to the original issue but just a sort of remnant I guess.

![793C7696-DB0B-44B4-BE0A-C4095481B957](https://user-images.githubusercontent.com/53174718/161257223-1ab9440d-c039-4c43-b82e-8eebb8ea8e64.jpeg)
